flow/cmd/maintenance.go

```diff
@@ -0,0 +1,250 @@
+package cmd
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"log/slog"
+	"os"
+
+	"github.com/aws/smithy-go/ptr"
+	"go.temporal.io/sdk/client"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/insecure"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"github.com/PeerDB-io/peer-flow/generated/protos"
+	"github.com/PeerDB-io/peer-flow/peerdbenv"
+	"github.com/PeerDB-io/peer-flow/shared"
+	peerflow "github.com/PeerDB-io/peer-flow/workflows"
+)
+
+type MaintenanceCLIParams struct {
+	TemporalHostPort                  string
+	TemporalNamespace                 string
+	Mode                              string
+	FlowGrpcAddress                   string
+	SkipIfK8sServiceMissing           string
+	FlowTlsEnabled                    bool
+	SkipOnApiVersionMatch             bool
+	SkipOnNoMirrors                   bool
+	UseMaintenanceTaskQueue           bool
+	AssumeSkippedMaintenanceWorkflows bool
+}
+
+type StartMaintenanceResult struct {
+	SkippedReason *string `json:"skippedReason,omitempty"`
+	APIVersion    string  `json:"apiVersion,omitempty"`
+	CLIVersion    string  `json:"cliVersion,omitempty"`
```

Comment:
not sure if `CLIVersion` is the best name?

Reply:
yeah, this is the local version of the CLI which is running the maintenance mode